integration_tests/tests/integration/simulation/mod.rs

```diff
@@ -0,0 +1,116 @@
+//! This integration test attempts to implement discrete event simulation in Janus. The goal of this
+//! test is to uncover more bugs around garbage collection, network errors, runtime errors, or other
+//! failures to uphold certain DAP invariants. We will try to avoid introducing nondeterminism as
+//! much as possible, given the current architecture, though we will necessarily fall short of
+//! complete determinism.
+//!
+//! The operating system, Postgres, and Rust synchronization primitives are outside the control of
+//! our simulation, so nondeterminism introduced by these sources is out of scope. While this set of
+//! tests may trigger bugs that involve these phenomena, it will not be able to reproduce them
+//! reliably like in-scope bugs. There will be room to fix some sources of nondeterminism short of
+//! these limits, in order to reproduce more bugs repeatably. For example, operations can be
+//! serialized, to avoid running multiple database transactions concurrently, and parallelism in the
+//! aggregation job driver and collection job driver can be eliminated.
+//!
+//! Prio3Histogram is the only VDAF used, and reports are carefully crafted to allow aggregate
+//! results to be verified, even in the face of partial data loss. Each report will have a unique
+//! measurement, so correct aggregate results must consist of only zeros and ones (and moreover,
+//! they should be all zeros for buckets that were never submitted in reports). A two in any
+//! position would indicate that a report was replayed or counted twice, while very large numbers
+//! would suggest an undetected batch mismatch between the leader and helper, or incorrect
+//! aggregation by one of the aggregators. There will only be one DAP task in use at a time. Both
+//! TimeInterval and FixedSize query types should be supported, as their implementations are very
+//! different.
+//!
+//! The simulation consists of multiple components making up a client, two aggregators, and a
+//! collector. All components for each aggregator share a database, and all components across the
+//! simulation share a `MockClock`. None of the components should run any asynchronous tasks
+//! continuously throughout the simulation (except for tokio-postgres connection tasks).
+//! Initialization will be akin to `JanusInProcessPair`, but more low-level. The simulation is fed a
+//! list of `Op` values, and it executes the operations described one after another. The operations
+//! are as follows:
+//!
+//! - AdvanceTime: Advance the `MockClock`'s time by some amount.
+//! - Upload: Have the client shard a report at the given timestamp, with the next sequential
+//!   measurement, and send it to the leader aggregator. The leader will handle the request and
+//!   store the report to the database. Note that, as currently implemented, this will wait for the
```

Comment:
These should be doccomments on the relevant enum variant. Otherwise there's a high chance we'll forget to ever upload this list.